runtime/src/main/java/org/corfudb/runtime/view/workflows/AddNode.java

```diff
@@ -55,4 +68,25 @@ protected boolean verifyRequest(@NonNull Layout layout) {
     public String toString() {
         return this.getClass().getSimpleName() + " " + nodeForWorkflow;
     }
+
+    @Override
+    protected Optional<ManagementClient> getOrchestrator() {
+        runtime.invalidateLayout();
+        Layout layout = new Layout(runtime.getLayoutView().getLayout());
+        BaseClient baseClient = runtime.getLayoutView().getRuntimeLayout(layout).getBaseClient(nodeForWorkflow);
+        // Add node requires a few pings for orchestrator selection because the current
+        // server router might not be immediately ready after the recent shutdown.
+        for (int i = 0; i < pingRetries; i++) {
+            if (baseClient.pingSync()) {
+                log.info("getOrchestrator: orchestrator selected {}", nodeForWorkflow);
+                return Optional.of(runtime.getLayoutView()
+                        .getRuntimeLayout(layout).getManagementClient(nodeForWorkflow));
+            }
+            Sleep.sleepUninterruptibly(pingInterval);
+            log.info("Retrying to ping a current base server {} times.", i);
+        }
+        log.warn("getOrchestrator: a server {} is not responding to pings", nodeForWorkflow);
+        return Optional.empty();
+
+    }
```

Comment:
This is true however, I need a different getOrchestrator() behavior based on the current  WorkflowRequest, e.g., I need a local one on the AddNode, HealNode and RestoreRedundnacy, and different one for the Remove workflows. I don't want to introduce another level of inheritance to make this happen, so it makes refactoring this a little tricky. I'm afraid it will be a big change that I need to think about if I want to refactor it properly. I think perhaps it better to leave it for this PR, and then refactor this in a separate PR.